packages/tldraw/src/lib/shapes/shared/TextLabel.tsx

```diff
@@ -110,25 +100,27 @@ export const TextLabel = React.memo(function TextLabel({
 					lineHeight: fontSize * lineHeight + 'px',
 					minHeight: lineHeight + 32,
 					minWidth: textWidth || 0,
-					color: isNote ? theme[labelColor].note.text : theme[labelColor].solid,
+					color: labelColor,
 					width: textWidth,
 					height: textHeight,
 				}}
 			>
 				<div className={`${cssPrefix} tl-text tl-text-content`} dir="ltr">
 					{finalText}
 				</div>
-				<TextArea
-					id={`text-input-${id}`}
-					ref={rInput}
-					// We need to add the initial value as the key here because we need this component to
-					// 'reset' when this state changes and grab the latest defaultValue.
-					key={initialText}
-					text={text}
-					isEditing={isEditing}
-					{...editableTextRest}
-					handleKeyDown={handleKeyDownCustom ?? editableTextRest.handleKeyDown}
-				/>
+				{(isEditingAnything || isSelected) && (
+					<TextArea
+						id={`text-input-${id}`}
+						ref={rInput}
+						// We need to add the initial value as the key here because we need this component to
+						// 'reset' when this state changes and grab the latest defaultValue.
+						key={initialText}
+						text={text}
+						isEditing={isEditing}
+						{...editableTextRest}
+						handleKeyDown={handleKeyDownCustom ?? editableTextRest.handleKeyDown}
+					/>
+				)}
```

Comment:
Our logic for showing the text area is:
- are we editing anything? -> yes
- are we selected? -> yes
- otherwise -> no

This lets us keep those elements out of the DOM until we need them.

Reply:
Makes sense to me, good call 👍